src/Analyser/MutatingScope.php

```diff
@@ -3466,23 +3470,37 @@ public function specifyExpressionType(Expr $expr, Type $type, Type $nativeType):
 					if ($dimType instanceof ConstantIntegerType) {
 						$types[] = new StringType();
 					}
+
+					// keep certainty
+					$variableCertainty = null;
+					if ($expr->var instanceof Variable && is_string($expr->var->name)) {
+						$variableCertainty = $scope->hasVariableType($expr->var->name);
```

Comment:
I tried using `hasExpressionType()` here, but it did not work for cases like

```
<?php declare(strict_types = 1);

use PHPStan\PhpDocParser\Ast\Type\GenericTypeNode;
use PHPStan\PhpDocParser\Ast\Type\IdentifierTypeNode;

class TypeNodeResolverRepro
{
	private function resolveGenericTypeNode(GenericTypeNode $typeNode): void
	{
		if ($typeNode->genericTypes[0] instanceof IdentifierTypeNode && $typeNode->genericTypes[0]->name === 'min') {
			$min = null;
		}

		if ($typeNode->genericTypes[1] instanceof IdentifierTypeNode && $typeNode->genericTypes[1]->name === 'max') {
		}
	}

}
```

which errored with

> Access to an undefined property PHPStan\PhpDocParser\Ast\Type\TypeNode::$name.

Reply:
fixed this in the latest commit. mistakenly re-used the `$certainty` variable within this if block, which overwrote the smae named parameter